### DISCUSSION TITLE
Mainboard cooling fan and part cooling fan linked?

### DISCUSSION THREAD

**Discussion by allylikesu** [bassamanator/Sovol-SV06-firmware#180] [General]
**Describe the bug**
I'm not sure if this is an inherent flaw in the SV06's design, or a wiring flaw, or a firmware flaw, but it seems that the mainboard cooling fan is linked to the part cooling fan. I noticed this when I was printing without the part cooling fan on, and got stepper driver overheating warnings in Mainsail interface.

**To Reproduce**
Turn the fan off in Mainsail's dashboard. Examine both part cooling fan and mainboard fan for movement.
Turn fan to 100% and check on both again.

**Expected behavior**
Mainboard fan is statically set to 100%, or automatically set based on stepper driver temps

**What printer are you using?**

SV06 with stock fans and mainboard

**What kind of device are you using to host Klipper?**

Pi 4B running MainsailOS

**How did you install Klipper?**

MainsailOS

**Is your local copy of this configuration up to date?**

I installed it yesterday.

**What slicer are you using?**

PrusaSlicer

**Are you using the config bundles from this repo?**

Yes

**Did you [Adjust your slicer](https://github.com/bassamanator/Sovol-SV06-firmware#adjust-your-slicer)**?

I used the provided config bundles

**Additional context**
I can't inspect the mainboard right now, as I have a print running. I have a desk fan pointed at the mainboard enclosure, which seems to be doing the trick for now.

**Comment by bassamanator** [comment]
By design? Who knows. Myself, I see it as a sign of neglect.

In any case, the behaviour that you observed is 'normal'.

The user @BeauSlim has a mod for the fans found [here](https://www.printables.com/@BeauSlim_355278).

**Comment by jumika** [comment]
@allylikesu You can also wire the fan to a fix 24v source on the board.

**Comment by alber70g** [comment]
Question regarding this: does the mainboard fan turn on when printing? Or also in pre-printing startup. I get a lot of times a `Stepper driver warning, OTPW Flag`.
I was wondering if the fan even turns on. With this airflow and the cooling parts it should stay cool enough, right?

![image](https://github.com/user-attachments/assets/56a72be3-2eaa-49ab-ab40-be7cc8884d80)

I suppose K-FAN1 and K-FAN2 are the case fans: https://www.sovol.tech/products/sovol-32-bit-silent-mainboard-sv06-sv06plus as reference

In the "Fans" section there's only one, 

https://github.com/bassamanator/Sovol-SV06-firmware/blob/master/printer.cfg#L172C1-L178C1

```
[fan]
pin: PA0
```

**Reply by bassamanator** [reply]
> the mainboard cooling fan is linked to the part cooling fan

[First paragraph](https://github.com/bassamanator/Sovol-SV06-firmware/discussions/180#discussion-7114627)

The warnings *might* go away after a klipper update.

**Reply by BeauSlim** [reply]
There is only one entry in the Klipper config because the K-FAN1 and K-FAN2 ports on the board are tied together electrically.  Their speed is controlled by a single pin from the CPU.  There is no way to fix this in settings alone.  You have to rewire.

**Reply by louia** [reply]
I’m also seeing an OTPW warning on stepper_y. Even with the fan set to 100%, the motherboard fan won’t spin, even if I manually nudge it. I don't know if it was the same on Merlin

**Reply by BeauSlim** [reply]
> I’m also seeing an OTPW warning on stepper_y. Even with the fan set to 100%, the motherboard fan won’t spin, even if I manually nudge it. I don't know if it was the same on Merlin

If you haven't already re-wired anything, and your part-cooling fan is at 100% but the MCU fan is off, then you likely have a loose connector or a bad case fan.
